plugins/huaweicloud/src/main/java/org/eclipse/xpanse/orchestrator/plugin/huaweicloud/monitor/HuaweiCloudMetricsService.java

```diff
@@ -0,0 +1,271 @@
+        this.huaweiCloudMonitorClient = huaweiCloudMonitorClient;
+        this.huaweiCloudMonitorCache = huaweiCloudMonitorCache;
+        this.huaweiCloudToXpanseDataModelConverter = huaweiCloudToXpanseDataModelConverter;
+    }
+
+    /**
+     * Get metrics of the @deployResource.
+     *
+     * @param resourceMetricRequest The request model to query metrics.
+     * @return Returns list of metric result.
+     */
+    public List<Metric> getMetricsByResource(ResourceMetricRequest resourceMetricRequest) {
+        List<Metric> metrics = new ArrayList<>();
+        DeployResource deployResource = resourceMetricRequest.getDeployResource();
+        CredentialDefinition credential = resourceMetricRequest.getCredential();
+        MonitorResourceType monitorResourceType = resourceMetricRequest.getMonitorResourceType();
+        clearExpiredMetricCache(deployResource.getResourceId());
+        ICredential icredential = getIcredential(credential);
+        CesClient client = huaweiCloudMonitorClient.getCesClient(icredential,
+                deployResource.getProperties().get("region"));
+        List<MetricInfoList> targetMetricInfoList =
+                getTargetMetricInfoList(deployResource, monitorResourceType,
+                        client);
+        for (MetricInfoList metricInfoList : targetMetricInfoList) {
+            ShowMetricDataRequest showMetricDataRequest =
+                    huaweiCloudToXpanseDataModelConverter.buildShowMetricDataRequest(
+                            resourceMetricRequest, metricInfoList);
+            ShowMetricDataResponse showMetricDataResponse =
+                    client.showMetricData(showMetricDataRequest);
+            Metric metric =
+                    huaweiCloudToXpanseDataModelConverter.convertShowMetricDataResponseToMetric(
+                            deployResource, showMetricDataResponse, metricInfoList);
+            metrics.add(metric);
+            cacheMetricData(metric, metrics, deployResource, metricInfoList);
+        }
+        return metrics;
+    }
+
+    private void cacheMetricData(Metric metric, List<Metric> metrics, DeployResource deployResource,
+                                 MetricInfoList metricInfoList) {
+        if (!CollectionUtils.isEmpty(metric.getMetrics())) {
+            metrics.add(metric);
+            huaweiCloudMonitorCache.set(deployResource.getResourceId(), metric);
+        } else {
+            List<Metric> metricCache =
+                    huaweiCloudMonitorCache.get(deployResource.getResourceId(),
+                            metricInfoList.getMetricName());
+            metrics.addAll(metricCache);
+        }
+    }
+
+    /**
+     * Get metrics of the @deployService.
+     *
+     * @param serviceMetricRequest The request model to query metrics.
+     * @return Returns list of metric result.
+     */
+    public List<Metric> getMetricsByService(ServiceMetricRequest serviceMetricRequest) {
+        List<Metric> metrics = new ArrayList<>();
```

Comment:
we can create this object in the convertBatchListMetricResponseToMetric method itself.

Reply:
Okay, I have made the changes.